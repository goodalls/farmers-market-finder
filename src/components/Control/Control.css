@import url("https://fonts.googleapis.com/css?family=Open+Sans:400,700,800");
.control {
  grid-column: 1/6;
  grid-row: 2;
  height: 100vh; }

.form {
  display: flex;
  flex-direction: column;
  align-items: center; }

input {
  text-align: center;
  background-color: transparent;
  border-radius: 10px;
  border: 2px solid #9C7B5C;
  color: #d97426;
  font-size: 1.5em;
  height: 50px;
  margin: 20px 10px;
  width: 50%; }

input[type='number']::-webkit-inner-spin-button,
input[type='number']::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0; }
