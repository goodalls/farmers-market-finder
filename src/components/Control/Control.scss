@import '../../styles/colors.scss';

.control {
  grid-column: 1/6;
  grid-row: 2;
  height: 100vh;
  text-align: center;
}

.form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

input {
  background-color: transparent;
  border-radius: 10px;
  border: 2px solid $primary;
  color: $primary-orange;
  font-size: 1.5em;
  height: 50px;
  margin: 20px 10px;
  width: 50%;
  text-align: center;
  outline:none;
}

input::placeholder {
  color: $primary-orange;
}

input[type='number']::-webkit-inner-spin-button,
input[type='number']::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.control h4 {
  font-size: 1.2em;
  font-weight: 700;
  margin: 20px 10px;
}

button {
  background-color: transparent;
  border-radius: 10px;
  border: 2px solid $primary;
  color: $primary-orange;
  font-size: 1.5em;
  height: 100px;
  margin: 20px 10px;
  width: 50%;
  outline:none;
}

.zip {
  width: 30%;
}
