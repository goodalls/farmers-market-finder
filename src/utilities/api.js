export const fetchParse = async url => {
  try {
    const initialFetch = await fetch(url);
    const response = await initialFetch.json();
    return response;
  } catch (error) {
    // throw new error('fetchParse error');
    return 'fetchParse error';
  }
};

export const marketDetails = async id => {
  try {
    const initialFetch = await fetchParse(`https://search.ams.usda.gov/farmersmarkets/v1/data.svc/mktDetail?id=${id}`);
    return initialFetch;
  } catch (error) {
    return 'marketDetails error';
  }
};
