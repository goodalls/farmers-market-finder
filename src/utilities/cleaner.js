export const cleanMarkets = array => {
  return array
    .map(market => {
      const distance = market.marketname.substr(0, 4);
      const marketname = market.marketname.substr(4);

      return { distance, marketname, id: market.id };
    })
    .slice(0, 10);
};
