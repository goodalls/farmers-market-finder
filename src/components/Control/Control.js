import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './Control.css';

export class Control extends Component {
  constructor(props) {
    super(props);
    this.state = {
      zip: '',
      searchByZip: false,
      searchByLocation: false
    };
  }

  handleChange = event => {
    const { value } = event.target;
    this.setState({ zip: value });
  };

  handleSubmit = event => {
    event.preventDefault();
    console.log('handleSubmit');
    this.setState({
      searchByZip: true,
      searchByLocation: false
    });
  };

  handleCurrentLocation = event => {
    this.setState({ 
      searchByZip: false, 
      searchByLocation: true 
    });
  };

  render() {
    return (
      <section className="control">
        <button onClick={this.handleCurrentLocation}>
          Search by Current Location
        </button>
        <h4>
          search for Farmers markets near you by entering your zip code below or
          choosing to find markets close to your current location
        </h4>
        {/* <NavLink to={'/sign-in'}>{Sign - In}</NavLink>
        <NavLink to={'/favorites'}>{Favorites}</NavLink> */}
        <form onSubmit={this.handleSubmit} className="form">
          <label htmlFor="zip" />
          <input
            type="number"
            id="zip"
            maxLength="5"
            name="zip"
            value={this.state.zip}
            onChange={this.handleChange}
            placeHolder="ZIP"
          />
          <input type="submit" />
        </form>
      </section>
    );
  }
}

export default Control;
