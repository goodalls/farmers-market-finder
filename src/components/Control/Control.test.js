import React from 'react';
import { Control, mapDispatchToProps } from './Control';
import { shallow } from 'enzyme';
import * as api from '../../utilities/api';

import * as cleaner from '../../utilities/cleaner';

describe('CONTROL', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Control />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('should set state of the entered zip code', () => {
      const wrapper = shallow(<Control />);
      const mockEvent = { target: { value: 80021 } };
      const expected = 80021;
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('zip')).toEqual(expected);
    });
  });

  describe('handleSubmit', () => {
    it('should set state call preventDefault', () => {
      const wrapper = shallow(<Control markets={jest.fn()}/>);
      const mockEvent = { target: { value: 80021 }, preventDefault: jest.fn() };
      wrapper.instance().handleSubmit(mockEvent);
      expect(mockEvent.preventDefault).toHaveBeenCalled();
    });

    it('should set state Zip = ""', () => {
      const wrapper = shallow(<Control markets={jest.fn()}/>);
      const mockEvent = { target: { value: 80021 }, preventDefault: jest.fn() };
      const expected = '';
      wrapper.setState({zip: 80021});
      wrapper.instance().handleSubmit(mockEvent);
      expect(wrapper.state('zip')).toEqual(expected);
    });
  });

  describe('handleCurrentLocation', () => {
    // it('should set state searchByZip = false', () => {
    //   const wrapper = shallow(<Control />);
    //   const mockEvent = { target: { value: 80021 }, preventDefault: jest.fn() };
    //   const expected = false;
    //   wrapper.instance().handleCurrentLocation(mockEvent);
    //   expect(wrapper.state('searchByZip')).toEqual(expected);
    // });
    // it('should set state searchByLocation = true', () => {
    //   const wrapper = shallow(<Control />);
    //   const mockEvent = { target: { value: 80021 }, preventDefault: jest.fn() };
    //   const expected = true;
    //   wrapper.instance().handleCurrentLocation(mockEvent);
    //   expect(wrapper.state('searchByLocation')).toEqual(expected);
    // });
  });

  describe('getNearbyMarkets', () => {
    it('should call api.fetchParse and cleaner.cleanMarkets', async() => {
      const wrapper = shallow(<Control />);
      const mockLat = 103.5;
      const mockLong = 1024;
      const expected = 'http://search.ams.usda.gov/farmersmarkets/v1/data.svc/locSearch?lat=103.5&lng=1024';
      api.fetchParse = jest.fn().mockReturnValue({});
      cleaner.cleanMarkets = jest.fn();
      await wrapper.instance().getNearbyMarkets(mockLat, mockLong);
      expect(api.fetchParse).toHaveBeenCalledWith(expected);
      expect(cleaner.cleanMarkets).toHaveBeenCalled();
   
    });
  });

  describe('MSTP and MDTP', () => {
    it('should define props', () => {});

    it('should call the dispatch function on MDTP', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      mapped.markets();
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});
