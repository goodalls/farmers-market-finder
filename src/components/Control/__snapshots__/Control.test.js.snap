// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CONTROL should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Control
    history={
        Object {
            "push": [Function],
          }
    }
    markets={[Function]}
    marketsArray={Array []}
    zipMarketsArray={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(User) />,
        <Link
          replace={false}
          to="/market-list/current"
>
          <button
                    onClick={[Function]}
          >
                    Search by Current Location
          </button>
</Link>,
        <h4>
          Search for Farmers Markets near you by entering your zip code below or choosing to find markets close to your current location
</h4>,
        <form
          className="form"
          onSubmit={[Function]}
>
          <label
                    htmlFor="zip"
          />
          <input
                    className="zip"
                    id="zip"
                    maxLength="5"
                    name="zip"
                    onChange={[Function]}
                    placeholder="ZIP"
                    type="number"
                    value=""
          />
          <input
                    className="submit"
                    type="submit"
                    value="Search by ZIP"
          />
</form>,
      ],
      "className": "control",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            onClick={[Function]}
>
            Search by Current Location
</button>,
          "replace": false,
          "to": "/market-list/current",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search by Current Location",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Search by Current Location",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search for Farmers Markets near you by entering your zip code below or choosing to find markets close to your current location",
        },
        "ref": null,
        "rendered": "Search for Farmers Markets near you by entering your zip code below or choosing to find markets close to your current location",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="zip"
/>,
            <input
              className="zip"
              id="zip"
              maxLength="5"
              name="zip"
              onChange={[Function]}
              placeholder="ZIP"
              type="number"
              value=""
/>,
            <input
              className="submit"
              type="submit"
              value="Search by ZIP"
/>,
          ],
          "className": "form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "htmlFor": "zip",
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "zip",
              "id": "zip",
              "maxLength": "5",
              "name": "zip",
              "onChange": [Function],
              "placeholder": "ZIP",
              "type": "number",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "submit",
              "type": "submit",
              "value": "Search by ZIP",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(User) />,
          <Link
            replace={false}
            to="/market-list/current"
>
            <button
                        onClick={[Function]}
            >
                        Search by Current Location
            </button>
</Link>,
          <h4>
            Search for Farmers Markets near you by entering your zip code below or choosing to find markets close to your current location
</h4>,
          <form
            className="form"
            onSubmit={[Function]}
>
            <label
                        htmlFor="zip"
            />
            <input
                        className="zip"
                        id="zip"
                        maxLength="5"
                        name="zip"
                        onChange={[Function]}
                        placeholder="ZIP"
                        type="number"
                        value=""
            />
            <input
                        className="submit"
                        type="submit"
                        value="Search by ZIP"
            />
</form>,
        ],
        "className": "control",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              onClick={[Function]}
>
              Search by Current Location
</button>,
            "replace": false,
            "to": "/market-list/current",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search by Current Location",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Search by Current Location",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search for Farmers Markets near you by entering your zip code below or choosing to find markets close to your current location",
          },
          "ref": null,
          "rendered": "Search for Farmers Markets near you by entering your zip code below or choosing to find markets close to your current location",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="zip"
/>,
              <input
                className="zip"
                id="zip"
                maxLength="5"
                name="zip"
                onChange={[Function]}
                placeholder="ZIP"
                type="number"
                value=""
/>,
              <input
                className="submit"
                type="submit"
                value="Search by ZIP"
/>,
            ],
            "className": "form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "htmlFor": "zip",
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "zip",
                "id": "zip",
                "maxLength": "5",
                "name": "zip",
                "onChange": [Function],
                "placeholder": "ZIP",
                "type": "number",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "submit",
                "type": "submit",
                "value": "Search by ZIP",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
